docs/content/ChoiceFieldset.mdx

```diff
@@ -0,0 +1,407 @@
+  </ChoiceFieldset.List>
+</ChoiceFieldset>
+```
+
+### Example: With pre-selected choices
+
+```jsx live
+<ChoiceFieldset selected={['figma', 'react']}>
+  <ChoiceFieldset.Legend>Prefered Primer component interface</ChoiceFieldset.Legend>
+
+  <ChoiceFieldset.List selectionVariant="multiple">
+    <ChoiceFieldset.Item value="figma">Figma library</ChoiceFieldset.Item>
+    <ChoiceFieldset.Item value="css">Primer CSS</ChoiceFieldset.Item>
+    <ChoiceFieldset.Item value="react">Primer React components</ChoiceFieldset.Item>
+    <ChoiceFieldset.Item value="viewcomponents">Primer ViewComponents</ChoiceFieldset.Item>
+  </ChoiceFieldset.List>
+</ChoiceFieldset>
+```
+
+### Example: with validation
+
+```javascript live noinline
+const choices = [
+  {
+    value: 'figma',
+    displayName: 'Figma library',
+    description: 'The Figma file where we have Figma symbols and light documentation'
+  },
+  {
+    value: 'css',
+    displayName: 'Primer CSS',
+    description: 'The OG. A CSS library with utility styles and component styles'
+  },
+  {
+    value: 'react',
+    displayName: 'Primer React components',
+    description: 'The React component library that these docs belong to'
+  },
+  {
+    value: 'viewcomponents',
+    displayName: 'Primer ViewComponents',
+    description: 'The Rails and Web Components implementation of our components'
+  }
+]
+
+const ChoiceFieldsetWithValidation = () => {
+  const [selectedChoices, setSelectedChoices] = React.useState([])
+  const [validationResult, setValidationResult] = React.useState()
+
+  React.useEffect(() => {
+    if (selectedChoices.length && selectedChoices.length > 2) {
+      setValidationResult('tooManySelections')
+    } else {
+      setValidationResult(undefined)
+    }
+  }, [selectedChoices])
+
+  return (
+    <ChoiceFieldset
+      name="radioGroup"
+      onSelect={selectedValues => {
+        setSelectedChoices(selectedValues)
+      }}
+      validationMap={{tooManySelections: 'error'}}
+      validationResult={validationResult}
+      selected={selectedChoices}
+    >
+      <ChoiceFieldset.Legend>Prefered Primer component interface</ChoiceFieldset.Legend>
+      <ChoiceFieldset.Description>Pick your top two</ChoiceFieldset.Description>
+
+      <ChoiceFieldset.List selectionVariant="multiple">
+        {choices.map(choice => (
+          <ChoiceFieldset.Item value={choice.value}>
+            <Item.Label>{choice.displayName}</Item.Label>
+            <Item.Caption>{choice.description}</Item.Caption>
+          </ChoiceFieldset.Item>
+        ))}
+      </ChoiceFieldset.List>
+
+      <ChoiceFieldset.Validation validationKey="tooManySelections">
+        Only two selections are allowed
+      </ChoiceFieldset.Validation>
+    </ChoiceFieldset>
+  )
+}
+
+render(<ChoiceFieldsetWithValidation />)
+```
+
+## ChoiceFieldset.Legend
+
+A title for the set of choices. A `ChoiceFieldset.Legend` must be passed, but it may be visually hidden.
```

Comment:
Technically, a legend is optional in a fieldset: 
> Permitted content An optional <legend> element, followed by flow content.

—https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset#technical_summary



I think maybe we could let consumers use a ChoiceFieldset without a legend. What do you think @alliethu ?